src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -449,6 +459,16 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                     responseStream = EmptyReadStream.Instance;
                     ReturnConnectionToPool();
                 }
+                else if (request.Method == s_httpConnectMethod && (int)response.StatusCode == 200)
```

Comment:
Also, I realize we're not consistent about this currently, but it's probably better to use the enum here, i.e. response.StatusCode == HttpStatusCode.Ok (or whatever it is)